contracts/InterchainProposalExecutor.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Ownable} from '@openzeppelin/contracts/access/Ownable.sol';
+import {AxelarExecutable} from '@axelar-network/axelar-gmp-sdk-solidity/contracts/executable/AxelarExecutable.sol';
+import {IAxelarExecutable} from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarExecutable.sol';
+import {StringToAddress} from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/AddressString.sol';
+import {InterchainProposalExecutorBase} from './executor/InterchainProposalExecutorBase.sol';
+import {InterchainCalls} from './lib/InterchainCalls.sol';
+
+/**
+ * @title InterchainProposalExecutor
+ * @dev This contract provides a simple implementation of the `InterchainProposalExecutorBase` abstract contract.
+ * It offers specific logic for handling proposal execution success and failures as well as emitting events
+ * after proposal execution.
+ */
+contract InterchainProposalExecutor is InterchainProposalExecutorBase {
```

Comment:
Can you merge this and the Base contract? The _on* methods can still be virtual with a default implementation so they could be overridden if anyone wants, but less contracts this way.